### PR TITLE
[Enhancement] Adds method_parameters in neural search query to support ef_search 

### PR DESCRIPTION
**Pull request by shatejas**
### Description
Enables efsearch and nprobes for neural search

merging from feature branch
Original approved PR: https://github.com/opensearch-project/neural-search/pull/787

### Check List
- [x] New functionality includes testing.
    - [x] All tests pass
- [ ] New functionality has been documented.
    - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by vibrantvarun — APPROVED** [review]
As the original PR in feature/ef-search branch is already approved. I am approving this one as well as everything looks good.

**Comment by vibrantvarun** [comment]
There is some issue in bwc I ran it locally and the same error which are coming on CI are also coming on local.

File: src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java [hunk 1]
```diff
@@ -69,18 +75,11 @@ public class NeuralQueryBuilder extends AbstractQueryBuilder<NeuralQueryBuilder>
     @VisibleForTesting
     static final ParseField QUERY_IMAGE_FIELD = new ParseField("query_image");
 
-    @VisibleForTesting
-    static final ParseField MODEL_ID_FIELD = new ParseField("model_id");
+    public static final ParseField MODEL_ID_FIELD = new ParseField("model_id");
```
Comment (vibrantvarun) [inline src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java:78]: Why it is public?

**Review by navneet1v — APPROVED** [review]

**Comment by navneet1v** [comment]
Merging this Pr as this PR is causing other main branch code to break.

**Pull request merged**